### PR TITLE
Mac Installer & Uninstaller: Fix display of name of helper apps

### PR DESCRIPTION
**Pull request by CharlieFenton**
Mac Installer & Uninstaller:
* Fix display of name of helper apps under MacOS 13 Ventura
* Fix installing a branded build over unbranded or an unbranded build over branded.

When a Mac is configured with multiple users, the BOINC installer and uninstaller temporarily install a small helper app for each user who is not currently logged in, and creates a LaunchAgent for each of those users. The next time the user logs in, the LaunchAgent runs the helper app, which completes the install or uninstall process for that  user and then deletes itself and its associated LaunchAgent. 

Instead of the name of the helper app (_BOINC_Finish_Install_ or _BOINC_Finish_Uninstall_), MacOS 13 Ventura incorrectly displayed the name of the developer who code signed it. Currently, that is my name. Because the developer's name is not familiar to the users, [it can cause them concern](https://boinc.berkeley.edu/forum_thread.php?id=14919#111066). This fixes that issue, which was a side effect of my changes from PR #5115.

It also fixes several other small issues I discovered while testing the changes for this PR, primarily when a user has unbranded (standard) _BOINCManager_ installed and then installs a branded version to replace it (for example, _World Community Grid_.) Also in the reverse case, such as when installing _BOINCManager_ to replace a branded version such as _World Community Grid_.

### CODE REVIEW

File: clientgui/mac/templates/FinishInstall-Info.plist [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+<plist version="1.0">
+<dict>
+	<key>CFBundleDevelopmentRegion</key>
+	<string>English</string>
+	<key>CFBundleIdentifier</key>
+	<string>edu.berkeley.boinc.finish-install</string>
+	<key>CFBundleInfoDictionaryVersion</key>
+	<string>6.0</string>
+	<key>CFBundleName</key>
+	<string> boinc cleanup</string>
```
Comment (AenBleidd) [inline clientgui/mac/templates/FinishInstall-Info.plist:12]: @CharlieFenton, did you put space at the beginning by intention or it's just a typo?

**Comment by CharlieFenton** [comment]
> @CharlieFenton, did you put space at the beginning by intention or it's just a typo?

Just a typo, though probably no consequence. I fixed it.

I'm working on fixing the CI build failure.

**Comment by CharlieFenton** [comment]
@AenBleidd Since I only changed Mac-specific files, I believe the CI errors for Android and Webassembly are not due to anything in this PR. This PR is ready for merging once the MacOS CI builds succeed.

**Comment by AenBleidd** [comment]
@CharlieFenton, yes, Android and WebAssembly are expected to fail now.
I'll re-review it and merge if everything else is fine.
Thank you for the fix.

**Review by AenBleidd — APPROVED** [review]

**Pull request merged**